### PR TITLE
RPC Timeout

### PR DESCRIPTION
**Pull request by Maithem**
Make the RPC timeout configurable via the CorfuRuntime.

Addresses https://github.com/CorfuDB/CorfuDB/issues/995

### CODE REVIEW

**Comment by Maithem** [comment]
Hmm, maybe I should address https://github.com/CorfuDB/CorfuDB/issues/991 before checking this in.

File: runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java [hunk 1]
```diff
@@ -145,6 +145,13 @@
     @Setter
     public int writeRetry = 3;
 
+    /**
+     * Set the timeout for rpc requests
+     */
+    @Getter
+    @Setter
+    public long timeoutResponse = 5000;
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java]: Make this a duration

File: runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java [hunk 2]
```diff
@@ -235,14 +235,14 @@ public NettyClientRouter(String host, Integer port, Boolean tls,
     public NettyClientRouter(String host, Integer port, Boolean tls,
                              String keyStore, String ksPasswordFile, String trustStore,
                              String tsPasswordFile, Boolean saslPlainText, String usernameFile,
-                             String passwordFile, MetricRegistry metricRegistry) {
+                             String passwordFile, MetricRegistry metricRegistry, long responseTimeout) {
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java]: Can we avoid adding more fields to pass information (the current amount is kind of crazy).
Make a ConnectionSettings (or similar class) to pass this data....

Comment (Maithem) [reply runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java]: Yea, its pretty bad. Will address #991 first.

**Comment by no2chem** [comment]
Superseded by #1090.

**Pull request closed without merging**